src/networkd.c

```diff
@@ -1546,58 +1550,40 @@ _netplan_networkd_write_wait_online(const NetplanState* np_state, const char* ro
          * come online.
          */
         if (!(def->optional || def->activation_mode)) {
+            WaitOnlineData* d = g_malloc0(sizeof(WaitOnlineData));
+            d->ignore_carrier = def->ignore_carrier;
+
             // Check if we have any IP configuration
             // bond and bridge members will never ask for link-local addresses (see above)
```

Comment:
thought: I wonder if we really should wait for all the bond members by default. One might have a bunch of the bond members disconnected for whatever reason and the system might still be online if at least one member is connected. I'd say it would be better to use `--any` for groups of interfaces that are members of the same bond. That would complicate the implementation though. I suppose we at least need to recommend setting all the members as `optional` (but not the bond itself) to avoid having anything blocking on the wait-online target...

Reply:
Good thought! I'm now issuing an INFO log message about it, at least.